### ISSUE TITLE
If using new AMQP connection, unable to send D2C messages after upgrading device SDK 1.6.0 to latest 1.18.0

### ISSUE THREAD

**Issue by anhashia** [Azure/azure-iot-sdk-csharp#620]
- **OS, version, SKU and CPU architecture used:** <VERSION> 
Windows 10 Desktop x64

- **Application's .NET Target Framework :** <NETVERSION> 
net452
- **Device:** <DEVICETYPE> 
Laptop
- **SDK version used:** <VERSION> 
SDK 1.18.0 (latest)

## Description of the issue:

D2C cloud message send attempt times out after couple of successful sends 
1.) Older version 1.6.0 doesn't show the issue
2.) Issue specific to AMQP and AMQP over WebSocket. It doesn't happen with MQTT
3.) Happens when new connection is created for each send

## Code sample exhibiting the issue:

private const string DeviceConnectionString = "HostName=MyHub.azure-devices.net;SharedAccessKeyName=iothubowner;SharedAccessKey=***********"; 
private static String deviceId = "mydevice1";
 static void Main(string[] args)
        {
            {

            
                    for (int i = 0; i < 5; i++)
                    {
                        Console.WriteLine(DateTime.UtcNow + " Sending msg " + i + " to " + DeviceConnectionString.Split(';')[0]);
                        SendMessageAsync(DeviceConnectionString, deviceId).Wait();
                        Console.WriteLine(DateTime.UtcNow + " Message sent\n----------------------\n");
                    }
                }
            }
 
  private static async Task SendMessageAsync(string connectionString, string deviceId)
        {
           
            
          using (DeviceClient deviceClient = DeviceClient.CreateFromConnectionString(connectionString, deviceId, TransportType.Amqp_Tcp_Only))
            {
                await deviceClient.SendEventAsync(new Message());
            }
  }

##  Console Log
```
Failure if using 1.18.0 (unable to send msg 2)
 
9/11/2018 11:05:18 PM Sending msg 0 to HostName=MyHub.azure-devices.net
9/11/2018 11:05:19 PM Message sent
----------------------
 
9/11/2018 11:05:19 PM Sending msg 1 to HostName=MyHub.azure-devices.net
9/11/2018 11:05:19 PM Message sent
----------------------
 
9/11/2018 11:05:19 PM Sending msg 2 to HostName=MyHub.azure-devices.net (gets stuck and times out )
```

**Comment by anhashia** [comment]
Issue specific to **AMQP** and AMQP over WebSocket. It **doesn't** happen with MQTT

**Comment by anhashia** [comment]
Similar to https://github.com/Azure/azure-iot-sdk-csharp/issues/580 . Only AMQP impacted.

**Comment by abhipsaMisra** [comment]
The current LTS version supports the behavior of creating device client and sending telemetry using the IoTHub connection string; so it can be used to unblock this issue. That being said, we highly encourage you to use the device connection string for creating the device client.

**Comment by phramusca** [comment]
For information, using SAS tokens does not work neither and ends up with timeouts as well
https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-security#use-a-shared-access-policy

`ITransportSettings[] transportSettings = {
                new AmqpTransportSettings(TransportType.Amqp_Tcp_Only)
                {
                    AmqpConnectionPoolSettings = new AmqpConnectionPoolSettings
                    {
                        Pooling = true,
                        MaxPoolSize = 1
                    }
                }
            };

            string hostName = "myHub.azure-devices.net";
            string password = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx";
            var sasRule = new SharedAccessSignatureBuilder
            {
                Key = password,
                KeyName = "device",
                Target = hostName + "/devices/",
                TimeToLive = TimeSpan.FromSeconds(60)
            };
            var authMethod = new DeviceAuthenticationWithToken(deviceId, sasRule.ToSignature());

            using (DeviceClient deviceClient = DeviceClient.Create(hostName, authMethod, transportSettings))
            {
                deviceClient.OperationTimeoutInMilliseconds = 4000;
                await deviceClient.SendEventAsync(new Message());
            }`

**Comment by anhashia** [comment]
Hi @phramusca
Yeah this  code snippet will also show the issue as all  3 conditions are met
1.) Using AMQP protocol
2.) Using Hub level primary key for SAS token generation and  not using device key which will be unique to each device.
3.) Disposing the object repeatedly in each call by calling using. Use https://github.com/Azure/azure-functions-host/wiki/Managing-Connections for reference. 

If you avoid doing #3, you can fix (workaround)  the problem. The other way to workaround the issue will be to use device specific primary key for SAS token generation (#2).

**Comment by anhashia** [comment]
Fixed in **1.20.0**. Unable to reproduce it on **1.20.0**

**Comment by az-iot-builder-01** [comment]
@abhipsaMisra, @phramusca, thank you for your contribution to our open-sourced project! Please help us improve by filling out this 2-minute [customer satisfaction survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbRx8jpG79OudPqY-BojcKjkJUMFFBQjgxNkw5ODJIRjBWTzI0Nk85R0pUQiQlQCN0PWcu)
